src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -137,6 +137,39 @@ public DependencyGraphSpec WithProjectClosure(string projectUniqueName)
             return projectDependencyGraphSpec;
         }
 
+        /// <summary>
+        /// Creates a new <see cref="DependencyGraphSpec" /> from a project name and project closure.
+        /// </summary>
+        /// <param name="projectUniqueName">The project's unique name.</param>
+        /// <param name="closure">The project's closure</param>
+        /// <returns>A <see cref="DependencyGraphSpec" />.</returns>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="projectUniqueName" />
+        /// is <c>null</c> or an empty string.</exception>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="closure" /> is <c>null</c>.</exception>
+        public DependencyGraphSpec CreateFromClosure(string projectUniqueName, IReadOnlyList<PackageSpec> closure)
+        {
+            if (string.IsNullOrEmpty(projectUniqueName))
+            {
+                throw new ArgumentException(Strings.ArgumentNullOrEmpty, nameof(projectUniqueName));
+            }
+
+            if (closure == null)
+            {
+                throw new ArgumentNullException(nameof(closure));
+            }
+
+            var dgSpec = new DependencyGraphSpec();
```

Comment:
Should the new dgSpec have the same _isReadOnly flag as the current dgSpec?

Reply:
It didn't before this change, so no.

Reply:
I believe that `_isReadOnly` is not enforced, so it's up to the caller to determine by their own usage whether it's read only or not. It's not good API design, but it gave big performance gains without a huge refactor, so it's what we have to live with.